src/utils/components/SelectTypeahead/SelectTypeahead.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { FC, Ref, useRef, useState } from 'react';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { interfacesTypes } from '@kubevirt-utils/resources/vm/utils/network/constants';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  Label,
+  MenuToggleElement,
+  Select,
+  SelectList,
+  SelectOption,
+  SelectOptionProps,
+} from '@patternfly/react-core';
+
+import Toggle from './Toggle';
+import { filterOptions } from './utils';
+
+type SelectTypeaheadProps = {
+  options: SelectOptionProps[];
+  placeholder: string;
+  selected: string;
+  setSelected: (newSelection: null | string) => void;
+};
+
+const SelectTypeahead: FC<SelectTypeaheadProps> = ({
+  options,
+  placeholder,
+  selected,
+  setSelected,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  const [inputValue, setInputValue] = useState<string>(selected || '');
+  const [focusedItemIndex, setFocusedItemIndex] = useState<null | number>(null);
+  const textInputRef = useRef<HTMLInputElement>();
+
+  const selectOptions = inputValue ? filterOptions(options, inputValue) : options;
+
+  const onSelect = (value: string) => {
+    if (value) {
+      setSelected(selected === value ? null : value);
+
+      if (selected === value) setInputValue('');
+
+      setIsOpen(true);
+    }
+
+    textInputRef.current?.focus();
+  };
+
+  return (
+    <Select
+      toggle={(toggleRef: Ref<MenuToggleElement>) => (
+        <Toggle
+          focusedItemIndex={focusedItemIndex}
+          inputValue={inputValue}
+          isOpen={isOpen}
+          onSelect={onSelect}
+          placeholder={placeholder}
+          selected={selected}
+          selectOptions={selectOptions}
+          setFocusedItemIndex={setFocusedItemIndex}
+          setInputValue={setInputValue}
+          setIsOpen={setIsOpen}
+          setSelected={setSelected}
+          textInputRef={textInputRef}
+          toggleRef={toggleRef}
+        />
+      )}
+      id="typeahead-select"
```

Comment:
if its going to be a generic component , so please use id with some prefix/suffix for generic purpose.